src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/hooks/index.ts

```diff
@@ -0,0 +1,58 @@
+import { IInstanceContext, IDataSources } from "altinn-shared/types";
+import { replaceTextResourceParams } from "altinn-shared/utils";
+import { buildInstanceContext } from "altinn-shared/utils/instanceContext";
+import { useState, useEffect } from "react";
+import { shallowEqual } from "react-redux";
+import { useAppSelector } from "src/common/hooks";
+import { IMapping, IOptionSource, IOption } from "src/types";
+import { getOptionLookupKey, getRelevantFormDataForOptionSource, replaceOptionDataField } from "src/utils/options";
+
+export const useGetOptions = (optionsId: string, mapping?: IMapping, source?: IOptionSource) => {
```

Comment:
In general I would prefer to accept an option object instead of separate parameters. That will make it easier to understand what parameters are passed (since they will be named), it makes it easy to skip optional parameters, and it makes it easier to extend the API in a nice way, since the order of "parameters" dont matter. [This article ](https://www.freecodecamp.org/news/elegant-patterns-in-modern-javascript-roro-be01e7669cbd/) also gives some more examples of benefits if you are interested :) 

f.ex
```jsx
const optionsId = 1;
const source = {
	group: 'group',
	label: 'label',
	value: 'value',
};
useGetOptions({optionsId, source});
```
vs
```jsx
const optionsId = 1;
const source = {
	group: 'group',
	label: 'label',
	value: 'value',
};
useGetOptions(optionsId, null, source)
```

Reply:
Thank you for the article, really interesting stuff. Completely on board. Let's use this for new stuff where it adds value and refactor old code that could benefit from this pattern when we touch it. 

Refactored ❤️